src/main/java/com/example/WeatherDataCodeChallenge/SensorController.java

```diff
@@ -26,70 +26,56 @@
     // Creates new sensor if sensorId number doesn't already exist or is null.
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public Sensor registerNewSensor (@RequestBody Sensor sensor){
-        if(sensor.getSensorId() == null)
-        {
+    public Sensor registerNewSensor(@RequestBody Sensor sensor) {
+        if (sensor.getSensorId() == null) {
+            String newSensorId = UUID.randomUUID().toString();
+            sensor.setSensorId(newSensorId);
             return sensorService.addNewSensor(sensor);
         }
-        else if(!sensorService.exists(sensor.getSensorId()))
-        {
+        else if(sensorService.findById(sensor.getSensorId()) == null){
             return sensorService.addNewSensor(sensor);
         }
-        else{
+        else {
             log.info("Sensor ID number taken");
             throw new IllegalStateException("Index is out of bounds");
         }
     }
 
-    public Sensor registerNewMetrics (@RequestBody Sensor sensor){
-        return sensorService.addNewSensor(sensor);
-    }
-
-    // Need to create new instance of sensorId/cityName/countryName with updated metricId/metrics/time
-    @PostMapping(path = "/{id}")
+    //POST for adding metrics values to a sensor
+    @PostMapping(path = "/{sensorId}")
```

Comment:
If it's just updating the sensor and not creating a new object in your DB, the HTTP method should probably be PUT/PATCH instead of POST